pyhealth/datasets/base_dataset.py

```diff
@@ -482,29 +524,32 @@ def task_stat(self) -> str:
         num_visits = len(set([sample["visit_id"] for sample in self.samples]))
         lines.append(f"\t- Number of visits: {num_visits}")
         lines.append(
-            f"\t- Number of visits per patient: {len(self) / num_patients:.4f}")
+            f"\t- Number of visits per patient: {len(self) / num_patients:.4f}"
+        )
         for key in self.samples[0]:
             if key in ["patient_id", "visit_id"]:
                 continue
-            if type(self.samples[0][key]) in [bool, int, str]:
-                num_events = [1 for sample in self.samples]
-            # a list of values
-            elif type(self.samples[0][key][0]) in [bool, int, str]:
-                num_events = [len(sample[key]) for sample in self.samples]
-            # a list of lists of values
-            elif type(self.samples[0][key][0]) == list:
-                num_events = [len(sum(sample[key], [])) for sample in self.samples]
+            # list
+            if type(self.samples[0][key]) == list:
+                nested = [isinstance(e, list) for s in self.samples for e in s[key]]
+                if any(nested):
+                    num_events = [len(sum(sample[key], [])) for sample in self.samples]
+                else:
```

Comment:
`else` may mean that samples have different nested levels.

Reply:
It can either be 1 or 2 level list as we check this in the end to set_task.

Reply:
I mean what if `nested=[True, False, True, False]` ?

Reply:
https://github.com/sunlabuiuc/PyHealth/blob/27ddaa7254ba94fea47b9a27d0cfcefdbdc21be9/pyhealth/datasets/base_dataset.py#L363-L370

This case is eliminated at the end of `set_task()`.